shared/src/api/index.coffee

```diff
@@ -165,13 +165,16 @@ class APIHandlerBase
 
     onSuccess ?= DEFAULT_SUCCESS
     onFail ?= DEFAULT_FAIL
-    return new Promise (resolve) =>
+    return new Promise (resolve, reject) =>
+
       delay =>
         @_xhr.request(requestConfig)
+          .then(((resp) -> resolve(resp); resp), ((resp) -> reject(resp); Promise.reject(resp)))
```

Comment:
Not really - the issue i was working around is that if it comes last then the functions only receive the results of the `.then` block before them, and if the final `rejection` doesn't bother calling `Promise.reject` then the error handler is never called.  That in turn caused the "loading" animation on the preview cards to spin forever. 

That's why I was careful to return the same thing that was passed to then `.then`  Of course there's still the danger that the handler could modify the `resp` and throw off the followup handlers.